crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <openssl/crypto.h>
+#include <openssl/service_indicator.h>
+
+#if defined(AWSLC_FIPS)
+int awslc_fips_service_indicator_get_counter(void) {
+  struct fips_service_indicator_state *indicator =
+      CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  if (!indicator) {
+    if(!awslc_fips_service_indicator_init_state()) {
+      return 0;
+    }
+    indicator = CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  }
+  return indicator->counter;
+}
+
+int awslc_fips_service_indicator_get_serviceID(void) {
+  struct fips_service_indicator_state *indicator =
+      CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  if (!indicator) {
+    if(!awslc_fips_service_indicator_init_state()) {
+      return 0;
+    }
+    indicator = CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  }
+  return indicator->serviceID;
+}
+
+int awslc_fips_service_indicator_reset_state(void) {
```

Comment:
why do we need the reset function?

Reply:
I guess you are referring to the power-on self-tests being done in FIPS mode? We should definitely reset the indicator at the end of the power-on tests function. Otherwise we'd be indicating incorrectly that the module provided a FIPS approved service to the consumer even though the services are internally called and not asked for by the consumer.

If that's done then you won't need to reset the indicator in the unit test. Why do you think this function would be useful to a consumer? I'd prefer to keep the exported APIs minimal (only what's necessary).